packages/modules/couchbase/src/couchbase-container.ts

```diff
@@ -0,0 +1,599 @@
+
+              throw new Error(message);
+            },
+            this.startupTimeout || 60_000
+          );
+        } else {
+          log.info(
+            `Primary index creation for bucket ${bucket.getName()} ignored, since QUERY service is not present.`
+          );
+        }
+      }
+    }
+  }
+
+  protected override async containerStarted(
+    startedTestContainer: StartedTestContainer,
+    inspectResult: InspectResult
+  ): Promise<void> {
+    const client = await getContainerRuntimeClient();
+
+    await this.waitUntilNodeIsOnline(client, inspectResult, startedTestContainer);
+    await this.initializeIsEnterprise(startedTestContainer);
+    await this.renameNode(startedTestContainer, inspectResult);
+    await this.initializeServices(startedTestContainer);
+    await this.setMemoryQuotas(startedTestContainer);
+    await this.configureAdminUser(startedTestContainer);
+    await this.configureExternalPorts(startedTestContainer);
+    if (this.enabledServices.has(CouchbaseService.INDEX)) {
+      await this.configureIndexer(startedTestContainer);
+    }
+
+    await Promise.all(
+      this.constructWaitStrategies().map((s) =>
+        s.waitUntilReady(
+          client.container.getById(startedTestContainer.getId()),
+          BoundPorts.fromInspectResult(client.info.containerRuntime.hostIps, inspectResult)
+        )
+      )
+    );
+
+    await this.createBuckets(client, inspectResult, startedTestContainer);
+
+    log.info(
+      `Couchbase container is ready! UI available at http://${startedTestContainer.getHost()}:${startedTestContainer.getMappedPort(
+        PORTS.MGMT_PORT
+      )}`
+    );
+  }
+
+  public override async start(): Promise<StartedCouchbaseContainer> {
+    const startingMessage = "Starting Couchbase Server -- Web UI available at http://<ip>:8091";
+    this.withExposedPorts(...(this.hasExposedPorts ? this.exposedPorts : this.getPortsToExpose())).withWaitStrategy(
+      Wait.forLogMessage(startingMessage)
+    );
+    const startedTestContainer = await super.start();
+
+    return new StartedCouchbaseContainer(startedTestContainer, this.username, this.password);
+  }
+}
+
+export class StartedCouchbaseContainer extends AbstractStartedContainer {
+  private readonly mappedKvPort: number;
+  private readonly host: string;
+  private readonly username: string;
+  private readonly password: string;
+
+  constructor(startedTestContainer: StartedTestContainer, username: string, password: string) {
+    super(startedTestContainer);
+    this.mappedKvPort = startedTestContainer.getMappedPort(PORTS.KV_PORT);
+    this.host = startedTestContainer.getHost();
+    this.username = username;
+    this.password = password;
+  }
+
+  getUsername() {
+    return this.username;
+  }
+
+  getPassword() {
+    return this.password;
+  }
+
+  getConnectionString() {
+    return `couchbase://${this.host}:${this.mappedKvPort}`;
```

Comment:
Better use getMappedPort in the getter, to get actual port, also when container was restarted.
```typescript
 return `couchbase://${this.host}:${startedTestContainer.getMappedPort(PORTS.KV_PORT)}`;
```